r/R/csv.R

```diff
@@ -623,9 +638,58 @@ readr_to_csv_convert_options <- function(na,
 #' @include arrow-package.R
 write_csv_arrow <- function(x,
                             sink,
+                            file = NULL,
                             include_header = TRUE,
-                            batch_size = 1024L) {
-  write_options <- CsvWriteOptions$create(include_header, batch_size)
+                            col_names = NULL,
+                            batch_size = 1024L,
+                            write_options = NULL,
+                            ...) {
+  unsupported_passed_args <- names(list(...))
+
+  if (length(unsupported_passed_args)) {
+    stop(
+      "The following ",
+      ngettext(length(unsupported_passed_args), "argument is ", "arguments are "),
+      "not yet supported in Arrow: ",
+      oxford_paste(unsupported_passed_args),
+      call. = FALSE
+    )
+  }
+
+  if (!missing(file) && !missing(sink)) {
+    stop(
+      paste("You have supplied both \"file\" and \"sink\" arguments. Please",
+            "supply only one of them."),
+      call. = FALSE
+    )
+  }
+
+  if (missing(sink) && !missing(file)) {
+    sink <- file
+  }
+
+  if (!missing(col_names) && !missing(include_header)) {
+    stop(
+      paste("You have supplied both \"col_names\" and \"include_header\"",
+            "arguments. Please supply only one of them."),
+      call. = FALSE
+    )
+  }
+
+  # default value are considered missing by base R
+  if (missing(include_header) && !missing(col_names)) {
+    message(
+      paste(
+        "You have supplied a value for \"col_names\". This will overwrite",
+        "the value for the \"include_headers\" argument."))
```

Comment:
`message()` pastes together its arguments (with no separator)
```suggestion
    message(
      "You have supplied a value for \"col_names\". This will overwrite ",
      "the value for the \"include_headers\" argument."
    )
```

Reply:
Done